Gems/ROS2/Code/Source/VehicleDynamics/DriveModels/SkidSteeringDriveModel.h

```diff
@@ -37,25 +37,43 @@ namespace ROS2::VehicleDynamics
         AZStd::pair<AZ::Vector3, AZ::Vector3> GetVelocityFromModel() override;
 
     private:
+        // Data collected once and cached for the reuse
+        struct SkidSteeringWheelData
+        {
+            WheelControllerComponent* wheelControllerComponentPtr{ nullptr };
+            WheelDynamicsData wheelData;
+            float wheelPosition{ 0.0f }; // normalized distance between the wheel and the axle's center
+            float dX{ 0.0f }; // wheel's contribution to vehicle's linear movement
+            float dPhi{ 0.0f }; // wheel's contribution to vehicle's rotational movement
+            AZ::Vector3 axis{ AZ::Vector3::CreateZero() }; // rotation axis of the wheel
+        };
+        AZStd::vector<SkidSteeringWheelData> m_wheelsCache;
+        bool m_initialized = false;
+
+        //! Collect all necessary data to compute the impact of wheels on the vehicle's velocity.
+        //! The data is stored in cache and reused.
+        void InitializeCache();
+
         //! Collect all necessary data to compute the impact of the wheel on the vehicle's velocity.
-        //! It can be thought of as a column of the Jacobian matrix of the mechanical system. Jacobian matrix for this model is a matrix of
-        //! size 2 x number of wheels. This function returns elements of column that corresponds to the given wheel and cache necessary data
-        //! to find the wheel's rotation as a scalar.
-        //! @param wheelNumber - number of wheels in axis
+        //! It can be thought of as a column of the Jacobian matrix of the mechanical system. Jacobian matrix for this model is a matrix
+        //! of size 2 x number of wheels. This function returns elements of column that corresponds to the given wheel and cache
+        //! necessary data to find the wheel's rotation as a scalar.
+        //! @param wheelId - id of the currently processed wheel in the axle vector
         //! @param axle - the wheel's axle configuration
-        //! @param axisCount - number of axles in the vehicle
-        //! @returns A tuple containing of :
-        //!  - pointer to WheelControllerComponent (API to query for wheel's ration speed),
+        //! @param axlesCount - number of axles in the vehicle
+        //! @returns A structure containing of:
```

Comment:
This documentation should link to the structure documentation, here just link to the structure.